server/http.go

```diff
@@ -53,11 +58,52 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
 		p.completeUserOAuthToZoom(w, r)
 	case "/deauthorization":
 		p.deauthorizeUser(w, r)
+	case postActionPath:
+		p.postActionConfirm(w, r)
 	default:
 		http.NotFound(w, r)
 	}
 }
 
+func (p *Plugin) postActionConfirm(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	response := model.PostActionIntegrationResponse{}
+	request := model.PostActionIntegrationRequestFromJson(r.Body)
+	accepted := request.Context[ContextAccept].(bool)
+	meetingID := request.Context[ContextMeetingID].(float64)
+
+	post := &model.Post{}
+	key := fmt.Sprintf("%v_%v", changeStatusKey, userID)
+
+	message := "Ok, the status won't be updated automatically"
+	changeStatus := no
+	if accepted {
+		changeStatus = yes
+		message = "You have accepted automatic status change. Yay!"
+	}
+
+	appErr := p.API.KVSet(key, []byte(changeStatus))
+	if appErr != nil {
+		p.API.LogDebug("failed to set status change preference", "error", appErr.Error())
+	}
```

Comment:
I think we should return here instead of proceeding after the if statement. Maybe have this function return an error, and call this function with a smaller function that handles the error. Any error that occurs in this function should cause the function to return early.

Reply:
@mickmister Here if we don't return early, the next call is to setUserStatus which would automatically send another slack attachment for user confirmation on automatic status change. Won't that be more convenient?

Also, if I understand this correctly, you are suggesting we remove the `KVSet` and `setUserStatus` snippets and put it in a separate function and return early if that function throws any error?